code/modules/spells/spell_types/hivemind.dm

```diff
@@ -0,0 +1,465 @@
+	var/mob/living/carbon/human/target = targets[1]
+	to_chat(user, "<span class='notice'>We begin increasing the psionic bandwidth between ourself and the vessel!</span>")
+	if(do_mob(user,user,60))
+		var/power = 120-get_dist(user, target)
+		if(power > 5 && user.z == target.z)
+			to_chat(target, "<span class='userdanger'>You feel a sharp pain, and foreign presence in your mind!!</span>")
+			to_chat(user, "<span class='notice'>We have overloaded the vessel for a short time!</span>")
+			target.Jitter(power)
+			target.adjustStaminaLoss(round(power/2))
+			target.Unconscious(power)
+		else
+			to_chat(user, "<span class='notice'>The vessel was too far away to be affected!</span>")
+	else
+		to_chat(user, "<span class='notice'>Our concentration has been broken!</span>")
+		revert_cast()
+
+/obj/effect/proc_holder/spell/self/hive_drain
+	name = "Repair Protocol"
+	desc = "Our many vessels sacrifice a small portion of their mind's vitality to cure us of our physical and mental ailments."
+
+	panel = "Hivemind Abilities"
+	charge_max = 600
+	clothes_req = 0
+	still_recharging_msg = "<span class='notice'>Our psionic powers are still recharging.</span>"
+	invocation_type = "none"
+	action_icon = 'icons/mob/actions/actions_hive.dmi'
+	action_background_icon_state = "bg_hive"
+	action_icon_state = "drain"
+	human_req = 1
+
+/obj/effect/proc_holder/spell/self/hive_drain/cast(mob/living/carbon/human/user)
+	var/datum/antagonist/hivemind/hive = user.mind.has_antag_datum(/datum/antagonist/hivemind)
+	if(!hive)
+		return
+	var/iterations = 0
+	var/power = 5
+
+	if(!user.getBruteLoss() && !user.getFireLoss() && !user.getCloneLoss() && !user.getBrainLoss())
+		to_chat(user, "<span class='notice'>We cannot heal ourselves any more with this power!</span>")
+		revert_cast()
+	to_chat(user, "<span class='notice'>We begin siphoning power from our many vessels!</span>")
+	while(iterations < 7)
+		var/mob/living/carbon/human/target = pick(hive.hivemembers)
+		if(!target)
+			break
+		if(!do_mob(user,user,15))
+			to_chat(user, "<span class='warning'>Our concentration has been broken!</span>")
+			break
+		target.adjustBrainLoss(5)
+		power = max(5-(round(get_dist(user, target)/40)),2)
+		if(user.getBruteLoss() > user.getFireLoss())
+			user.heal_ordered_damage(power, list(CLONE, BRUTE, BURN))
+		else
+			user.heal_ordered_damage(power, list(CLONE, BURN, BRUTE))
+		if(!user.getBruteLoss() && !user.getFireLoss() && !user.getCloneLoss()) //If we don't have any of these, stop looping
+			to_chat(user, "<span class='warning'>We finish our healing</span>")
+			break
+		iterations++
+	user.setBrainLoss(0)
+
+
+/obj/effect/proc_holder/spell/target_hive/hive_force
+	name = "Cerebellic Pulse"
+	desc = "We pulse the cerebellum of the target, forcing them to move in whatever direction we look at."
+	charge_max = 600
+	action_icon_state = "force"
+
+/obj/effect/proc_holder/spell/target_hive/hive_force/cast(list/targets, mob/living/user = usr)
```

Comment:
The control on this is pretty jank and I'm pretty sure targets can resist out of it at full HP while slowed targets are at its mercy